lib/datadog/ci/ext/environment.rb

```diff
@@ -6,48 +8,51 @@ module Ext
+        TAG_WORKSPACE_PATH = 'ci.workspace_path'
 
         PROVIDERS = [
-          ['APPVEYOR'.freeze, :extract_appveyor],
-          ['TF_BUILD'.freeze, :extract_azure_pipelines],
-          ['BITBUCKET_COMMIT'.freeze, :extract_bitbucket],
-          ['BUILDKITE'.freeze, :extract_buildkite],
-          ['CIRCLECI'.freeze, :extract_circle_ci],
-          ['GITHUB_SHA'.freeze, :extract_github_actions],
-          ['GITLAB_CI'.freeze, :extract_gitlab],
-          ['JENKINS_URL'.freeze, :extract_jenkins],
-          ['TEAMCITY_VERSION'.freeze, :extract_teamcity],
-          ['TRAVIS'.freeze, :extract_travis],
-          ['BITRISE_BUILD_SLUG'.freeze, :extract_bitrise]
+          ['APPVEYOR', :extract_appveyor],
+          ['TF_BUILD', :extract_azure_pipelines],
+          ['BITBUCKET_COMMIT', :extract_bitbucket],
+          ['BUILDKITE', :extract_buildkite],
+          ['CIRCLECI', :extract_circle_ci],
+          ['GITHUB_SHA', :extract_github_actions],
+          ['GITLAB_CI', :extract_gitlab],
+          ['JENKINS_URL', :extract_jenkins],
+          ['TEAMCITY_VERSION', :extract_teamcity],
+          ['TRAVIS', :extract_travis],
+          ['BITRISE_BUILD_SLUG', :extract_bitrise]
         ].freeze
 
         module_function
 
         def tags(env)
-          provider = PROVIDERS.find { |c| env.key? c[0] }
-          return {} if provider.nil?
-
-          tags = send(provider[1], env)
+          _, extractor = PROVIDERS.find { |provider_env_var, _| env.key?(provider_env_var) }
+          if extractor
+            tags = send(extractor, env)
 
-          tags[Datadog::Ext::Git::TAG_TAG] = normalize_ref(tags[Datadog::Ext::Git::TAG_TAG])
-          tags.delete(Datadog::Ext::Git::TAG_BRANCH) unless tags[Datadog::Ext::Git::TAG_TAG].nil?
-          tags[Datadog::Ext::Git::TAG_BRANCH] = normalize_ref(tags[Datadog::Ext::Git::TAG_BRANCH])
-          tags[Datadog::Ext::Git::TAG_REPOSITORY_URL] = filter_sensitive_info(tags[Datadog::Ext::Git::TAG_REPOSITORY_URL])
+            tags[Datadog::Ext::Git::TAG_TAG] = normalize_ref(tags[Datadog::Ext::Git::TAG_TAG])
+            tags.delete(Datadog::Ext::Git::TAG_BRANCH) unless tags[Datadog::Ext::Git::TAG_TAG].nil?
+            tags[Datadog::Ext::Git::TAG_BRANCH] = normalize_ref(tags[Datadog::Ext::Git::TAG_BRANCH])
+            tags[Datadog::Ext::Git::TAG_REPOSITORY_URL] = filter_sensitive_info(tags[Datadog::Ext::Git::TAG_REPOSITORY_URL])
 
-          # Expand ~
-          workspace_path = tags[TAG_WORKSPACE_PATH]
-          if !workspace_path.nil? && (workspace_path == '~' || workspace_path.start_with?('~/'))
-            tags[TAG_WORKSPACE_PATH] = File.expand_path(workspace_path)
+            # Expand ~
+            workspace_path = tags[TAG_WORKSPACE_PATH]
+            if !workspace_path.nil? && (workspace_path == '~' || workspace_path.start_with?('~/'))
+              tags[TAG_WORKSPACE_PATH] = File.expand_path(workspace_path)
+            end
+          else
+            tags = {}
           end
```

Comment:
Ha, this one is a mild spoiler alert for the next PR, that adds a fallback "provider".
I had both features implemented in my repo, but decided to split it into this PR first.

Trust me, it looks uglier now, but it will make reviewing the next one easier :)